src/components/PinInput/README.md

```diff
@@ -0,0 +1,312 @@
+				this.invalidEntry = false;
+			} else {
+				this.invalidEntry = true;
+				this.$refs.pinInput.shakeAndClearInputs();
+			}
+		},
+	},
+};
+</script>
+```
+
+### Outline variant
+```vue
+<template>
+	<m-pin-input
+		ref="pinInput"
+		:invalid="invalidEntry"
+		variant="outline"
+		@complete="onCodeComplete"
+	>
+		<template
+			v-if="invalidEntry"
+			#error
+		>
+			<m-notice type="error">
+				Error slot
+			</m-notice>
+		</template>
+	</m-pin-input>
+</template>
+
+<script>
+import { MPinInput } from '@square/maker/components/PinInput';
+import { MNotice } from '@square/maker/components/Notice';
+
+export default {
+	components: {
+		MPinInput,
+		MNotice,
+	},
+	data() {
+		return {
+			testCode: '123456',
+			invalidEntry: false,
+		};
+	},
+	methods: {
+		onCodeComplete(code) {
+			if (this.testCode === code) {
+				this.invalidEntry = false;
+			} else {
+				this.invalidEntry = true;
+				this.$refs.pinInput.shakeAndClearInputs();
+			}
+		},
+	},
+};
+</script>
+```
+
+### With mocked API latency
+```vue
+<template>
+	<m-pin-input
+		ref="pinInput"
+		variant="outline"
+		:invalid="invalidEntry"
+		:disabled="disableInput"
+		@complete="onCodeComplete"
+	>
+		<template
+			v-if="invalidEntry"
+			#error
+		>
+			<m-notice type="error">
+				Error slot
+			</m-notice>
+		</template>
+	</m-pin-input>
+</template>
+
+<script>
+import { MPinInput } from '@square/maker/components/PinInput';
+import { MNotice } from '@square/maker/components/Notice';
+
+export default {
+	components: {
+		MPinInput,
+		MNotice,
+	},
+	data() {
+		return {
+			testCode: '123456',
+			invalidEntry: false,
+			disableInput: false,
+		};
+	},
+	methods: {
+		onCodeComplete(code) {
+			this.disableInput = true;
+			const DELAY_MS = 500;
+
+			// simulate API time for code validation
+			setTimeout(() => {
+				if (this.testCode === code) {
+					this.invalidEntry = false;
+				} else {
+					this.invalidEntry = true;
+					this.$refs.pinInput.shakeAndClearInputs();
+				}
+				this.disableInput = false;
+			}, DELAY_MS);
+		},
+	},
+};
+</script>
+```
+
+### In a dialog for mobile testing
+```vue
+<template>
+	<div>
+		<m-button
+			size="small"
+			@click="openDialog"
+		>
+			Open dialog
+		</m-button>
+		<m-dialog-layer />
+	</div>
+</template>
+
+<script>
+import { MButton } from '@square/maker/components/Button';
+import { MDialogLayer } from '@square/maker/components/Dialog';
+import DemoDialog from 'doc/DemoDialog.vue';
+
+export default {
+	name: 'DemoSetup',
+
+	components: {
+		MDialogLayer,
+		MButton,
+	},
+
+	mixins: [
+		MDialogLayer.apiMixin,
+	],
+
+	methods: {
+		openDialog() {
+			this.dialogApi.open(() => <DemoDialog />);
+		},
+	},
+};
+</script>
+```
+
+_DemoDialog.vue_
+
+```vue
+<template>
+	<m-dialog>
+		<m-dialog-content>
+			<m-heading :size="3">
+				Dialog heading
+			</m-heading>
+			<m-pin-input
+				ref="pinInput"
+				:class="$s.padding"
+				:invalid="invalidEntry"
+				variant="outline"
+				@complete="onCodeComplete"
```

Comment:
can you rename the `onCodeComplete` function in the examples to `onPinComplete`?